### PR TITLE
Moved auth interceptors to common module

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Moved auth interceptors to common module, so that they can be used for different connector modules.
* Removed DataSourceSettings and merged in OpensearchSettings. This class handles updates as well.
* Fixed bug with OpensearchSettings where initially value set in opensearch.yml is not being picked up. 
    * For example in the below code block. PPL_ENABLED is always set to default value(true) even though it is set false in opensearch.yml.
    * ```
      public RestPPLQueryAction( Settings pluginSettings, org.opensearch.common.settings.Settings clusterSettings) {
      super();
      this.pplEnabled =
          () ->
              MULTI_ALLOW_EXPLICIT_INDEX.get(clusterSettings)
                  && (Boolean) pluginSettings.getSettingValue(Settings.Key.PPL_ENABLED);
       } 
* Introduced new setting `plugins.query.datasources.uri.allowhosts` for blocking URIs with unwanted hosts. This can protect from ssrf attacks.
* Made `plugins.query.datasources.encryption.masterkey` as final and can only be set in yml and can't be changed via 
API.
* Updated documentation can be found here: https://github.com/vamsi-amazon/sql/blob/7c3b5069fb71dea71cb1d2e4c28beb531a56b93e/docs/user/ppl/admin/datasources.rst#datasource-allow-hosts-config
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by anirudha** [comment]
Thanks

**Review by anirudha — DISMISSED** [review]

**Review by ps48 — DISMISSED** [review]

File: docs/user/ppl/admin/datasources.rst [hunk 1]
```diff
@@ -136,6 +136,17 @@ Master Key config for encrypting credential information
     # Print the master key
     print("Generated master key:", master_key)
 
+Datasource Allow Hosts Config
```
Comment (penghuo) [inline docs/user/ppl/admin/datasources.rst:139]: additional layer of security?

File: opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java [hunk 2]
```diff
@@ -123,6 +146,10 @@ public OpenSearchSettings(ClusterSettings clusterSettings) {
         METRICS_ROLLING_WINDOW_SETTING, new Updater(Key.METRICS_ROLLING_WINDOW));
     register(settingBuilder, clusterSettings, Key.METRICS_ROLLING_INTERVAL,
         METRICS_ROLLING_INTERVAL_SETTING, new Updater(Key.METRICS_ROLLING_INTERVAL));
+    register(settingBuilder, clusterSettings, Key.DATASOURCES_URI_ALLOWHOSTS,
+        DATASOURCE_URI_ALLOW_HOSTS, new Updater(Key.DATASOURCES_URI_ALLOWHOSTS));
+    settingBuilder.put(Key.CLUSTER_NAME, ClusterName.CLUSTER_NAME_SETTING);
```
Comment (penghuo) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:151]: why SQL plugin handle cluster_name setting?

File: opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java [hunk 3]
```diff
@@ -98,6 +102,25 @@ public class OpenSearchSettings extends Settings {
       Setting.Property.NodeScope,
       Setting.Property.Dynamic);
 
+  // we are keeping this to not break upgrades if the config is already present.
+  // This will be completely removed in 3.0.
+  public static final Setting<InputStream> DATASOURCE_CONFIG = SecureSetting.secureFile(
```
Comment (penghuo) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:107]: why move to opensearch module? would it be better in datasource module?

**Review by penghuo — DISMISSED** [review]

File: opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java (hunk 2 above)
Comment (vamsi-amazon) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:151]: SQL Plugin requires this in StorageEngineFactory class in datasource connector modules.

File: opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java (hunk 3 above)
Comment (vamsi-amazon) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java:107]: Wanted to make OpenSearchSettings class a single interface for all settings. 
* This class is an extension for Setting class in commons module and available for all modules.
* This class has also got the implementation of consuming latest updates.

File: docs/user/ppl/admin/datasources.rst (hunk 1 above)
Comment (vamsi-amazon) [reply docs/user/ppl/admin/datasources.rst:139]: This is to give customer a config to disallow any kind of ssrf attacks.

**Review by ps48 — DISMISSED** [review]

**Review by penghuo — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Review by rupal-bq — APPROVED** [review]

**Pull request merged**